test/powershell/Modules/Microsoft.PowerShell.Utility/JsonObject.Tests.ps1

```diff
@@ -0,0 +1,45 @@
+﻿Describe 'Unit tests for JsonObject' -tags "CI" {
+
+    function ShouldThrow
+    {
+        param (
+            [Parameter(ValueFromPipeline = $true)]
+            $InputObject,
+            [Parameter(Position = 0)]
+            $expectedException
+        )
+
+        try
+        {
+            & $InputObject
+            throw "Should throw exception"
+        } catch
+        {
+            $_.FullyQualifiedErrorId | should be $expectedException
+        }
+    }
+
+    $validStrings = @(
+        @{ str = "" }
+        @{ str = "  " }
+        @{ str = "{a:1}" }
+    )
+
+    It 'no error for valid string' -TestCase $validStrings {
```

Comment:
It's better to make the test case name different for different test cases, so that when one of the test cases failed, you will know which test data hashtable caused the failure. The recommended way is:
```
$validStrings = @(
     @{ name = "empty"; str = "" }
     @{ name = "spaces"; str = "  " }
     @{ name = "{a:1}"; str = "{a:1}" }
)

It "no error for valid string - <name>" -TestCase $validStrings {
    ....
}
````